Sources/ProjectSpec/Scheme.swift

```diff
@@ -58,6 +59,36 @@ public struct Scheme: Equatable {
         }
     }
 
+    public struct StoreKitConfiguration: Equatable {
+        public var location: String
+        public var forWorkspace: Bool?
+
+        public init(location: String, forWorkspace: Bool? = nil) {
+            self.location = location
+            self.forWorkspace = forWorkspace
+        }
+
+        public init?(parentJSONDictionary: JSONDictionary, keyPath: KeyPath = "storeKitConfiguration") {
+            if let storeKitConfiguration: StoreKitConfiguration = parentJSONDictionary.json(atKeyPath: keyPath) {
+                // attempt to decode as StoreKitConfiguration
+                self = storeKitConfiguration
+            } else if let storeKitLocation: String = parentJSONDictionary.json(atKeyPath: keyPath) {
+                // attempt to decode as String
+                self = .init(location: storeKitLocation)
+            } else {
+                return nil
+            }
+        }
+
+        private var prefix: String {
+            return forWorkspace == true ? "../" : "../../"
```

Comment:
Are we sure the prefixes will never change or are other things in different scenarios? Maybe we should rename this to prefix, give it a default of "../../" and add info in the docs that it must be changed for workspaces. Thoughts?

Reply:
Personally I would have left has as a single `storekitConfiguration` string that required putting the prefix in and documented the usage of it. I'm hesitant to put too many smarts in XcodeGen for small side features, especially anything in Schemes as they change so much. But you've done the hard work here to make it more friendly already. If it's left in, it would be good to have some tests for the prefix changing with the boolean

Reply:
As XcodeGen gets older I have a greater appreciation for small features that don't need to updated when Xcode does 😄

Reply:
I absolutely see your point here, and I don't want to ship something that'll break with no recourse the next time an Xcode internal detail changes. How about this: instead of `forWorkspace`, I'll add `pathPrefix` with a default value of `../../`? `SimulateLocation` could also stand to benefit from this addition as it's currently broken for non-workspace projects...

Reply:
This way, the path can still be relative to the project root, same as other sources, but for `.xcworkspace` projects or when Xcode breaks, we can tweak as needed by setting `pathPrefix`? I'd just like for XcodeGen users not to have to worry about this weird wonky path detail until they actually need to

Reply:
@yonaskolb Ready for another round of review, does this look better?